cmd/cvp-trigger/main_test.go

```diff
@@ -1,55 +1,653 @@
+		Status       pjapi.ProwJobState
+		ArtifactsURL string
+		URL          string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    []byte
+		wantErr bool
+	}{
+		{
+			name: "",
+			fields: fields{
+				Status:       "success",
+				ArtifactsURL: "http://example.com/jobName/1/artifacts",
+				URL:          "http://example.com/jobName/1/",
+			},
+			want: []byte(`{
+    "status": "success",
+    "prowjob_artifacts_url": "http://example.com/jobName/1/artifacts",
+    "prowjob_url": "http://example.com/jobName/1/"
+}`),
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			p := &prowjobResult{
+				Status:       tt.fields.Status,
+				ArtifactsURL: tt.fields.ArtifactsURL,
+				URL:          tt.fields.URL,
+			}
+			got, err := p.toJSON()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("toJSON() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("toJSON() got = \n%v, want \n%v", string(got), string(tt.want))
 			}
-			if err != nil && !tc.expectError {
-				t.Errorf("%s: unexpected validation error: %v", tc.description, err)
+		})
+	}
+}
+
+func Test_getJobArtifactsURL(t *testing.T) {
+	org := "redhat-operator-ecosystem"
+	repo := "playground"
+	bucket := "origin-ci-test"
+	browserPrefix := "https://gcsweb-ci.svc.ci.openshift.org/gcs/"
+	jobName := "periodic-ci-redhat-operator-ecosystem-playground-cvp-ocp-4.4-cvp-common-aws"
+
+	prowConfig := &prowconfig.Config{
+		JobConfig: prowconfig.JobConfig{},
+		ProwConfig: prowconfig.ProwConfig{
+			Plank: config.Plank{
+				Controller: prowconfig.Controller{},
+				DefaultDecorationConfigs: map[string]*pjapi.DecorationConfig{
+					fmt.Sprintf("%s/%s",org,repo): {GCSConfiguration: &pjapi.GCSConfiguration{Bucket: bucket}},
+				},
+			},
+			Deck: prowconfig.Deck{
+				Spyglass: prowconfig.Spyglass{GCSBrowserPrefix: browserPrefix},
+			},
+		},
+	}
+	specWithRef := pjapi.ProwJobSpec{
+		Job:  jobName,
+		Refs: &pjapi.Refs{Org: org, Repo: repo},
+		Type: "periodic",
+	}
+	prowJobWithSpecRef := pjapi.ProwJob{
+		TypeMeta:   v1.TypeMeta{},
+		ObjectMeta: v1.ObjectMeta{Name: "jobWithSpecRef"},
+		Spec:       specWithRef,
+		Status:     pjapi.ProwJobStatus{State: "success", BuildID: "100"},
+	}
+	specWithExtraRef := pjapi.ProwJobSpec{
+		Job: jobName,
+		ExtraRefs: []pjapi.Refs{
+			{Org: org, Repo: repo},
+			{Org: "org2", Repo: "repo2"},
+		},
+		Type: "periodic",
+	}
+	prowJobWithExtraRef := &pjapi.ProwJob{
+		TypeMeta:   v1.TypeMeta{},
+		ObjectMeta: v1.ObjectMeta{Name: "jobWithExtraRef"},
+		Spec:       specWithExtraRef,
+		Status:     pjapi.ProwJobStatus{State: "success", BuildID: "101"},
+	}
+	type args struct {
+		prowJob *pjapi.ProwJob
+		config  *prowconfig.Config
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "Returns artifacts URL when we have .Spec.Ref",
+			args: args{
+				prowJob: &prowJobWithSpecRef,
```

Comment:
Why do we need named variables for `prowJobWithSpecRef` and `specWithRef`? Couldn't we just inline a `&ProwJob{}` literal in these test cases?